### PR TITLE
SA-1606 - Subscription limit on reports

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 -Tooltip shows if limit for reports reached
 - Button for creating new report disabled if limit for reports reached

### How To Test
 - Go to report builder
 - Check appteam
 - Create new report button should be working
 - Create new account
 - Enable user ui option `allow_saved_reports`
 - Button should be disabled
 - Tooltip should exist

### To Do
- [x] Check usage (since list isn't guaranteed to be the full list)

### Notes
 - Rebased off of SA-1588. Will mark appropriate file changes

### CODE REVIEW

File: cypress/fixtures/billing/subscription/200.get.reports-limited.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (PatrickSison) [inline cypress/fixtures/billing/subscription/200.get.reports-limited.json:1]: File added for subscription limit

File: src/pages/reportBuilder/ReportBuilder.js [hunk 2]
```diff
@@ -64,6 +69,19 @@ export function ReportBuilder({
     }
   }, [refreshSummaryReport, reportOptions, isEmpty]);
 
+  useEffect(() => {
```
Comment (PatrickSison) [inline src/pages/reportBuilder/ReportBuilder.js:72]: Retrieves subscription

File: src/pages/reportBuilder/ReportBuilder.js [hunk 3]
```diff
@@ -109,27 +127,35 @@ export function ReportBuilder({
...
     <Page
       title="Analytics Report"
-      primaryAction={
-        isSavedReportsEnabled
-          ? {
-              content: 'Save New Report',
-              onClick: () => setShowSaveNewReportModal(true),
-            }
-          : undefined
+      primaryArea={
```
Comment (PatrickSison) [inline src/pages/reportBuilder/ReportBuilder.js:130]: Tooltip and button disabling

File: cypress/fixtures/billing/subscription/200.get.json [hunk 4]
```diff
@@ -32,6 +32,11 @@
         "plan": "ip-0519",
         "product": "dedicated_ip",
         "limit": 4
+      },
```
Comment (PatrickSison) [inline cypress/fixtures/billing/subscription/200.get.json:35]: Added higher report limit for default subscription

File: src/pages/reportBuilder/ReportBuilder.js [hunk 5]
```diff
@@ -64,6 +69,19 @@ export function ReportBuilder({
     }
   }, [refreshSummaryReport, reportOptions, isEmpty]);
 
+  useEffect(() => {
+    getSubscription();
+  }, [getSubscription]);
+
+  const reportsProduct = subscription?.products?.find(({ product }) => product === 'reports');
```
Comment (kchu1337) [inline src/pages/reportBuilder/ReportBuilder.js:76]: I think you can move this within the memoization.

**Review by kchu1337 — CHANGES_REQUESTED** [review]
Can you also rebase from master so that only your 1606 commit shows up?

**Review by kchu1337 — COMMENTED** [review]
LGTM, can you just rebase onto master? It might not be necessary since we are squashing commits, but the PR would be cleaner.

**Review by kchu1337 — APPROVED** [review]

**Pull request merged**